src/Driver/IBMDB2/Connection.php

```diff
@@ -33,7 +33,7 @@ final class Connection implements ConnectionInterface
      *
      * @param resource $connection
      */
-    public function __construct(private $connection)
+    public function __construct(private readonly mixed $connection)
```

Comment:
Only typed properties may be flagged `readonly` and there is no native `resource` type. These are known limitations of the PHP engine. This is why I'm adding a `mixed` type here. Since the doc block is more precise, this change should not make a difference for static analysis.

Reply:
I don't really like using the `mixed` native type where the argument is of a more specific type but given that it's the limitation of the type system, it should be okay.